### PR TITLE
test: extend bootupd test to include testing adoption and updates

### PR DESCRIPTION
**Pull request by HuijingHei**
See https://github.com/coreos/fedora-coreos-tracker/issues/1788#issuecomment-2326473398

---
denylist: deny `ext.config.boot.bootupd-validate` until we rollout F41

### CODE REVIEW

**Comment by HuijingHei** [comment]
The adoption for `BIOS` is implemented with `bootupd >= 0.2.22`, but we do not have the new build yet.

**Comment by HuijingHei** [comment]
The CI will not run this test. Do testing manually based on `next-devel` image passed.
```
$ kola run  -E ./fedora-coreos-config --qemu-image data/fedora-coreos-41.20241017.10.0-qemu.x86_64.qcow2 ext.fedora-coreos-config.boot.bootupd-validate
=== RUN   ext.fedora-coreos-config.boot.bootupd-validate
--- PASS: ext.fedora-coreos-config.boot.bootupd-validate (58.27s)
PASS, output in _kola_temp/qemu-2024-10-18-0615-58
```

File: tests/kola/boot/bootupd-validate [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            ;;
+        *)
+            echo "skipped validate"
+            ;;
+    esac
+}
+
+overwrite_file
+if [ -n "${overwrite}" ] && bootupctl validate 2>&1; then
+    fatal "bootupctl validate failed as expected"
```
Comment (jbtrystram) [inline tests/kola/boot/bootupd-validate]: I think the message should be `"bootupctl validate did not fail as expected" ?

**Review by jbtrystram — DISMISSED** [review]
I am a bit unsure about the overwrite failure message, otherwise LGTM.

File: tests/kola/boot/bootupd-validate (hunk 1 above)
Comment (HuijingHei) [reply tests/kola/boot/bootupd-validate]: Agree with you, update the message, thanks!

File: kola-denylist.yaml [hunk 2]
```diff
@@ -17,3 +17,11 @@
...
     - next-devel
     - next
+- pattern: ext.config.boot.bootupd-validate
+  tracker: https://github.com/coreos/fedora-coreos-tracker/issues/1788
+  snooze: 2024-11-18
+  streams:
+    - stable
+    - testing
+    - testing-devel
+    - next
```
Comment (travier) [inline kola-denylist.yaml]: This should work on next now as it's running F41 and we've pushed the updated bootupd there.

**Comment by travier** [comment]
Otherwise looks good but I only gave it a quick look. Will take another look next week.

File: kola-denylist.yaml (hunk 2 above)
Comment (HuijingHei) [reply kola-denylist.yaml]: Current next is `41.20241006.1.1` which has old `bootupd-0.2.20-2.fc41`, but next-devel `41.20241017.10.0` which has new bootupd-0.2.24-1.fc41, so we will not run `next` until move to `next-devel`, is this right?

Comment (travier) [reply kola-denylist.yaml]: ~~Let's hold until we have the right version in next at least~~

Comment (travier) [reply kola-denylist.yaml]: Ignore my previous comment.

Comment (HuijingHei) [reply kola-denylist.yaml]: Sure, remove `next`, thanks!

**Comment by HuijingHei** [comment]
The test is passed using next `fedora-coreos-41.20241020.1.0-qemu.x86_64.qcow2`.
```
$ kola run  -E ./fedora-coreos-config --qemu-image data/fedora-coreos-41.20241020.1.0-qemu.x86_64.qcow2 ext.fedora-coreos-config.boot.bootupd-validate
=== RUN   ext.fedora-coreos-config.boot.bootupd-validate
--- PASS: ext.fedora-coreos-config.boot.bootupd-validate (58.43s)
PASS, output in _kola_temp/qemu-2024-10-22-0150-41
```

**Review by jmarrero — APPROVED** [review]
lgtm

**Comment by travier** [comment]
This fails in the RHCOS pipeline: https://github.com/openshift/os/pull/1644#issuecomment-2462331993

@HuijingHei can you take a look and either make something specific for RHCOS or keep the test only for FCOS for now? Thanks

**Pull request merged**